### DISCUSSION TITLE
TypeError: Unknown file extension ".css"

### DISCUSSION THREAD

**Discussion by Ramon404** [vitest-dev/vitest#6138] [Q&A]
### Describe the bug

Vitest is failing the tests due to it not being able to recognise a .css file:

`TypeError: Unknown file extension ".css" for ...\node_modules\@fontsource\assistant\200.css`

I have tried multiple "solutions" with no success, some include:
- setting up server.deps.inline
- mocking the @fontsource dependency

Here are my project dependencies:

- "react": "^18.2.0",
- "react-dom": "^18.2.0",
- "react-redux": "^9.1.2",
- "react-router-dom": "^6.23.1",
- "@testing-library/jest-dom": "^6.4.5",
- "@testing-library/react": "^16.0.0",
- "jsdom": "^24.1.0",
-  "vite": "^5.2.0",
-  "vitest": "^2.0.3"

### Used Package Manager

npm

**Comment by Ramon404** [comment]
I have resolved this issue by running `npm ls @fontsource/assistant` in my project to check which other dependencies are using this package. I then added all the relevant packages to `server.deps.inline`.

**Comment by freirepaulo** [comment]
I have the same problem here, and when I try to use the server.deps.inline doesn't work and highlights the deps with this error:
![Screenshot 2024-08-30 at 11 13 28](https://github.com/user-attachments/assets/c3e5b984-e08e-40c6-8e3c-18b309841e29)

**Reply by ILRamon404** [reply]
server.deps.inline is part of the test object e.g.

```
test: {
    server: {
        deps: {
            inline: [...]
        }
    }
}
```

**Comment by nilsbunger** [comment]
Setting test.server.deps.inline worked for me. Note that you have to include all packages in the dependency chain if the CSS error comes from a dependency of a dependency.
